tests/test_opds.py

```diff
@@ -1743,73 +1743,104 @@ def test_format_types(self):
-        eq_(feed.links[2].title, AudiobooksEntryPoint.INTERNAL_NAME)
-        eq_(feed.links[3].href, feed.link)
-        eq_(feed.links[3].title, "2011")
+        # The top level with no entrypoint
+        # Top Level Title >
+        feed = MockFeed()
+        feed.add_breadcrumbs(lane)
+        children = getElementChildren(feed)
+
+        eq_(len(children), 1)
+        eq_(children[0].attrib.get("href"), TestAnnotator.default_lane_url())
+        eq_(children[0].attrib.get("title"), TestAnnotator.top_level_title())
+
+        # The top level with an entrypoint
+        # Top Level Title > Audio
+        feed = MockFeed()
+        feed.add_breadcrumbs(lane, entrypoint=ep)
+        children = getElementChildren(feed)
+
+        eq_(len(children), 2)
+        eq_(children[0].attrib.get("href"), TestAnnotator.default_lane_url())
+        eq_(children[0].attrib.get("title"), TestAnnotator.top_level_title())
+        eq_(children[1].attrib.get("href"), TestAnnotator.default_lane_url() + "?entrypoint=" + ep.URI)
+        eq_(children[1].attrib.get("title"), ep.INTERNAL_NAME)
+
+        # One lane level down but with no entrypoint
+        # Top Level Title > 2001
+        feed = MockFeed()
+        feed.add_breadcrumbs(sublane)
+        children = getElementChildren(feed)
+
+        eq_(len(children), 2)
+        eq_(children[0].attrib.get("href"), TestAnnotator.default_lane_url())
+        eq_(children[0].attrib.get("title"), TestAnnotator.top_level_title())
+        assert(("?entrypoint=" + ep.URI) not in children[1].attrib.get("href"))
+        eq_(children[1].attrib.get("title"), "2001")
+
+        # One lane level down and with an entrypoint
+        # Each sublane will have the entrypoint propagated down to its link
+        # Top Level Title > Audio > 2001
+        feed = MockFeed()
+        feed.add_breadcrumbs(sublane, entrypoint=ep)
+        children = getElementChildren(feed)
+
+        eq_(len(children), 3)
+        eq_(children[0].attrib.get("href"), TestAnnotator.default_lane_url())
+        eq_(children[0].attrib.get("title"), TestAnnotator.top_level_title())
+        eq_(children[1].attrib.get("href"), TestAnnotator.default_lane_url() + "?entrypoint=" + ep.URI)
+        eq_(children[1].attrib.get("title"), ep.INTERNAL_NAME)
+        assert(("?entrypoint=" + ep.URI) in children[2].attrib.get("href"))
+        eq_(children[2].attrib.get("title"), "2001")
+
+        # Two lane levels down but no entrypoint
+        # Top Level Title > 2001 > 2002
+        feed = MockFeed()
+        feed.add_breadcrumbs(subsublane)
+        children = getElementChildren(feed)
+
+        eq_(len(children), 3)
+        eq_(children[0].attrib.get("href"), TestAnnotator.default_lane_url())
+        eq_(children[0].attrib.get("title"), TestAnnotator.top_level_title())
+        assert(("?entrypoint=" + ep.URI) not in children[1].attrib.get("href"))
+        eq_(children[1].attrib.get("title"), "2001")
```

Comment:
You can't rely on the names of objects you create with the helper functions -- the test might change slightly and then this would be "2002" instead of "2001". Instead, check this against the variable containing the value, e.g. `sublane.title`.